official/transformer/transformer_main.py

```diff
@@ -182,6 +182,11 @@ def get_train_op_and_metrics(loss, params):
     if params["use_tpu"] and params["tpu"] != tpu_util.LOCAL:
       optimizer = tf.contrib.tpu.CrossShardOptimizer(optimizer)
 
+    # Uses automatic mixed precision FP16 training if on GPU.
+    if params["dtype"] == "fp16" and not params["use_tpu"]:
```

Comment:
I don't think the use_tpu part is necessary. fp16 isn't support on TPUs, so users should not pass --dtype=fp16 with --use_tpu.